sql/2024/02/median-image-bytes.sql

```diff
@@ -0,0 +1,71 @@
+# HTTP Archive query to get median image size in MB per image format.
+#
+# WPP Research, Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# See **TODO**
+
+SELECT
+  format,
+  APPROX_QUANTILES(bytesImg, 1000)[
+    OFFSET
+      (500)] / 1024 / 1024 AS median_img_mbytes
+FROM (
+       SELECT
+         DISTINCT url,
+                  info AS version
+       FROM
+         `httparchive.technologies.2024_01_01_mobile`
+       WHERE
+           app = 'WordPress')
+       JOIN (
+  SELECT
+    url,
+    format,
+    bytesImg
+  FROM (
+         SELECT
+           pageid,
+           format,
+         FROM
+           `httparchive.summary_requests.2024_01_01_mobile`
+         WHERE
+             format IN ("webp",
+                        "png",
+                        "gif",
+                        "jpg",
+                        "avif")
+         GROUP BY
+           pageid,
+           format)
+         JOIN (
+    SELECT
+      pageid,
+      url,
+      bytesImg
+    FROM
+      `httparchive.summary_pages.2024_01_01_mobile`)
+              USING
```

Comment:
the joins would be clearer if you used named tables first then joined then, eg. "WITH table_name AS ( Query... ), another_table AS ( Query... )"

Reply:
+1, it's much easier to read/review individual queries using the `WITH` operator, and then you would just reference the other query results in the JOINs, rather than nesting the entire queries.

To give a quick example, it could look something like this:
```
WITH wpUrls AS (
  SELECT
    date,
    client,
    page
  FROM
    `httparchive.all.pages`
  WHERE
    IS_CMS(technologies, 'WordPress', '')
),

imageRequests AS (
  SELECT
    date,
    client,
    page,
    ...
  FROM
    `httparchive.all.requests`
  WHERE
    ... # Ensure this is only images.
)

SELECT
  ...
FROM
  wpUrls
INNER JOIN
  imageRequests
USING
  (date, client, page)
...
```